log10/load.py

```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
     return completionID
 
 
+class StreamingResponseWrapper:
+    def __init__(self, completion_url, completionID, response, partial_log_row):
+        self.completionID = completionID
+        self.completion_url = completion_url
+        self.partial_log_row = partial_log_row
+        self.response = response
+        self.final_result = ""  # Store the final result
+        self.start_time = time.perf_counter()
+        self.gpt_id = None
+        self.model = None
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            chunk = next(self.response)
+            content = chunk.choices[0].delta.content
+            if content:
+                # Here you can intercept and modify content if needed
+                self.final_result += content  # Save the content
+                # Yield the original or modified content
+
+                self.model = chunk.model
+                self.gpt_id = chunk.id
+
+            return chunk
+        except StopIteration as se:
+            # Log the final result
+            # Create fake response for openai format.
+            response = {
+                "id": self.gpt_id,
+                "object": "completion",
+                "model": self.model,
+                "choices": [
+                    {
+                        "index": 0,
+                        "finish_reason": "stop",
```

Comment:
could max_token reached be the other reason?

Reply:
Let's see if we can get the stop reason from the last chunk above. Will give that a go

Reply:
Worked - should have been updated